lib/sprite_animation.dart

```diff
@@ -72,33 +73,30 @@ class SpriteAnimation {
   ///     Animation.sequenced('sheet.png', 8, textureY: 32.0 * i, textureWidth: 32.0, textureHeight: 32.0);
   /// This will create the i-th animation on the 'sheet.png', given it has 8 frames.
   SpriteAnimation.sequenced(
-    String imagePath,
+    Image image,
     int amount, {
     int amountPerRow,
     double textureX = 0.0,
     double textureY = 0.0,
     double textureWidth,
     double textureHeight,
     double stepTime = 0.1,
-    this.loop = true,
-  }) : assert(amountPerRow == null || amount >= amountPerRow) {
-    amountPerRow ??= amount;
-    frames = List<SpriteAnimationFrame>(amount);
-    for (var i = 0; i < amount; i++) {
-      final Sprite sprite = Sprite(
-        imagePath,
-        x: textureX + (i % amountPerRow) * textureWidth,
-        y: textureY + (i ~/ amountPerRow) * textureHeight,
-        width: textureWidth,
-        height: textureHeight,
-      );
-      frames[i] = SpriteAnimationFrame(sprite, stepTime);
-    }
-  }
+    bool loop = true,
+  }) : this.variableSequenced(
```

Comment:
I thought this name to be a bit confusing but I don't have a better one

Reply:
Me too, but alike you, I can't think on a better name, totally open to suggestions.

Reply:
Can you describe what it does and maybe we can come up with a new one together?

Reply:
I will address that in a followup. i have some ideas to improve these constructors now that we did this refactor.